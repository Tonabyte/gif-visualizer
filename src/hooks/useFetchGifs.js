import { useState, useEffect } from 'react';

// Services
import Services from '../services/services';

const useFetchGifs = category => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    Services.fetchGifs(category)
      .then(images => {
        setTimeout(() => {
          setState({
            data: images,
            loading: false,
          })
        }, 2000);
      });
  }, [category]);

  return state;
}

export default useFetchGifs;
