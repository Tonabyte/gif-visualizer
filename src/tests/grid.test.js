import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

// Components
import Grid from '../components/grid';

// Hooks
import useFetchGifs from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Grid Test Suite', () => {
  const props = {
    category: 'test',
  };

  it('Should show Grid component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<Grid { ...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('Should load items when images are loaded', () => {
    useFetchGifs.mockReturnValue({
      data: [{
        id: 0,
        title: 'title',
        image: 'image',
      }],
      loading: false,
    });
    const wrapper = shallow(<Grid { ...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
