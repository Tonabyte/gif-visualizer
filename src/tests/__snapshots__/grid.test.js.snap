// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Test Suite Should load items when images are loaded 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  Cargando
  <div
    className="cardGrid"
  >
    <GifCard
      id={0}
      key="0"
      title="title"
      url="url"
    />
  </div>
</Fragment>
`;

exports[`Grid Test Suite Should show Grid component 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  Cargando
  <div
    className="cardGrid"
  />
</Fragment>
`;
